### PR TITLE
Container improvements

### PR DESCRIPTION
**Pull request by elken**
Nothing earth-shattering here, changes documented in the commits but I'll summarize here as well as provide the necessary post-merge setup

## Clean up image

The qlot image that this is based on installs qlot to the root user only, which breaks when trying to run the container as a user other than root. The reason for alpine is it produces a smaller image (200MB smaller) and otherwise runs just fine. Happy to revert this commit if needed, the main reason for the PR is the second point

## Add support for GHCR

Since it's free, might as well take advantage of hosting an image on github so that users don't have to build their own and deal with the issues associated (like #1271). You can see the result of the build [on my fork](https://github.com/elken/lem/pkgs/container/lem) and the only bit of post-merge setup is go to [here](https://github.com/lem-project/lem/settings/actions) and change the below setting to reflect the image
image

Also not sure exactly what logic should determine what triggers a new image deployment, currently it'll run regardless of the CI building/failing but I can have it linked to that if needed; though it seems to be [failing](https://github.com/lem-project/lem/actions/workflows/test.yml)...

And to show that it works
image

If preferred, I can also look into pushing to docker hub instead; but I don't think there's much point outside of saving having to type `ghcr.io/`.

Thanks all!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
-RUN apt-get update && apt-get install gcc libncurses-dev -y
+RUN apk add --no-cache curl bash build-base ncurses-dev sbcl git
 
-RUN qlot install
+RUN curl -L https://qlot.tech/installer | bash
 
-RUN qlot exec sbcl --noinform --load scripts/build-ncurses.lisp
+RUN qlot install && \
+    qlot exec sbcl --noinform --load scripts/build-ncurses.lisp
```
Comment (cxxxr) [inline Dockerfile:12]: sbcl installed from apt is out of date, so we need to use fukamachi/qlot or fukamachi/sbcl

Comment (cxxxr) [reply Dockerfile:12]: also, thank you very much for the excellent PR.

Comment (elken) [reply Dockerfile:12]: Alpine, it only looks out by 1 minor version https://pkgs.alpinelinux.org/package/edge/community/x86_64/sbcl :)

Comment (cxxxr) [reply Dockerfile:12]: I didn't know that.
That's good then.

**Pull request merged**